server/svix-server/src/wait_for.rs

```diff
@@ -16,13 +16,31 @@ pub async fn wait_for_dsn(
     let sleep = tokio::time::sleep(Duration::from_secs(wait_for_seconds));
     tokio::pin!(sleep);
 
+    const ATTEMPT_TIMEOUT: Duration = Duration::from_secs(3);
+
+    let mut failed_attempt_rety_wait_ms: u64 = 125;
+    const MAX_RETRY_WAIT_MS: u64 = 3000;
+
     loop {
         tokio::select! {
-            res = tokio::net::TcpStream::connect((host.to_string(), port)) => {
-                if res.is_err() {
-                    continue;
-                } else {
-                    break;
+            res = tokio::time::timeout(ATTEMPT_TIMEOUT, tokio::net::TcpStream::connect((host.to_string(), port))) => {
+                match res {
+                    Ok(Ok(_)) => break,
+                    Ok(Err(_)) => {
```

Comment:
Going to make a comment that I don't know if it can be solved, but I wanted to bring this up: the code is very confusing at the moment. You have timeouts in three different places and the `Ok(Ok(_))` is indecipherable (should at least have comments explaining what which option is).

I don't know, but I wonder if it's possible to simplify it somehow, something like: an overall timeout enforced like the select is now, and then just wrap the TcpConnect with a timeout, so that every time we attempt this attempt has a timeout for itself (of e.g. 3s). Then after every attempt (or timeout) just sleep for 500ms, that's it. (it's looped)

Does that make sense?
```

Reply:
If you mean setting some TCP socket-level option: that's not possible with tokio `TcpStream`s, it's a bit antithetical with async. I can simplify by treating both the timeout-error `Err(_)` and the `Ok(Err(_))` in the same branch and just apply a fixed 500ms sleep.

Reply:
I more meant:
```rust
pub async fn andors_tcp_connect(
    dsn: &str,
) {
    tokio::select! {
        res = TcpConnect(...) => { return true }

       _request_timeout => { return false }
   }
}
```

And then just use this function as a timeoutable TCP request function.

You can then do in the main code that we already have:
```rust
if andors_tcp_connect(dsn) {
    break
} else {
    sleep(500ms).await
}
```
Get what I mean?


This will make the code very easy to parse and follow.

Reply:
I simplified it so that both the early-error and timeout cases are handled in one branch, and it uses a simple fixed retry.